Source/WebCore/testing/MockGamepadProvider.cpp

```diff
@@ -87,8 +95,14 @@ bool MockGamepadProvider::connectMockGamepad(unsigned index)
 
     m_connectedGamepadVector[index] = m_mockGamepadVector[index].get();
 
-    for (auto& client : m_clients)
+    for (auto& client : m_clients) {
         client->platformGamepadConnected(*m_connectedGamepadVector[index], EventMakesGamepadsVisible::Yes);
+        if (m_invisibleGamepadsForClient.contains(client)) {
+            for (auto& invisibleGamepad : m_invisibleGamepadsForClient.get(client))
```

Comment:
This is testing code so I don't care too much but this is inefficient. We're doing a hash map lookup 3 times here. Once in contains(), then in get() and once again in remove().